public/app/plugins/datasource/graphite/components/GraphiteVariableEditor.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+
+import { InlineField, Input, Select } from '@grafana/ui';
+
+import { GraphiteQuery, GraphiteQueryType } from '../types';
+
+import { convertToGraphiteQueryObject } from './helpers';
+
+interface Props {
+  query: GraphiteQuery | string;
+  onChange: (query: GraphiteQuery) => void;
+}
+
+const GRAPHITE_QUERY_VARIABLE_TYPE_OPTIONS = [
+  { label: 'Default Query', value: GraphiteQueryType.Default },
+  { label: 'Value Query', value: GraphiteQueryType.Value },
+  { label: 'Metric Name Query', value: GraphiteQueryType.MetricName },
+];
+
+export const GraphiteVariableEditor: React.FC<Props> = (props) => {
+  const { query, onChange } = props;
+  const [value, setValue] = useState(convertToGraphiteQueryObject(query));
+
+  return (
+    <>
+      <InlineField label="Select query type" labelWidth={20}>
+        <Select
+          aria-label="select query type"
+          options={GRAPHITE_QUERY_VARIABLE_TYPE_OPTIONS}
+          width={25}
+          value={value.queryType ?? GraphiteQueryType.Default}
+          onChange={(selectableValue) => {
+            onChange({
+              ...value,
+              queryType: selectableValue.value,
+            });
+          }}
+        />
+      </InlineField>
+      <InlineField label="Select query type" labelWidth={20} grow>
+        <Input
+          aria-label="Variable editor query input"
+          value={value.target}
+          onBlur={() => onChange(value)}
+          onChange={(e) => {
+            setValue({
+              ...value,
+              target: e.currentTarget.value,
+            });
+          }}
```

Comment:
Hey, what do you think about moving these functions on `onBlur` and `onChange` to live outside the render scope? It can create performance issues, once on each render, a new function will be created. 

PS: It also can be applied in the field above.